pkg/crane/options.go

```diff
@@ -47,26 +51,43 @@ func makeOptions(opts ...Option) Options {
 		},
 		Keychain: authn.DefaultKeychain,
 	}
+
 	for _, o := range opts {
 		o(&opt)
 	}
+
+	// Allow for untrusted certificates if the user passed Insecure but no custom
+	// transport.
+	if opt.insecure && !opt.transportSet {
+		transport := remote.DefaultTransport.(*http.Transport).Clone()
+		transport.TLSClientConfig = &tls.Config{
+			InsecureSkipVerify: true,
+		}
+		WithTransport(transport)(&opt)
+	}
+
 	return opt
 }
 
 // Option is a functional option for crane.
 type Option func(*Options)
 
 // WithTransport is a functional option for overriding the default transport
-// for remote operations.
+// for remote operations. Setting a transport will override the Insecure option's
+// configuration allowing for image registries to use untrusted certificates.
 func WithTransport(t http.RoundTripper) Option {
 	return func(o *Options) {
 		o.Remote = append(o.Remote, remote.WithTransport(t))
+		o.transportSet = true
 	}
 }
 
 // Insecure is an Option that allows image references to be fetched without TLS.
+// This will also allow for untrusted (e.g. self-signed) certificates in cases where
+// the default transport is used (i.e. when WithTransport is not defined).
```

Comment:
```suggestion
// the default transport is used (i.e. when WithTransport is not used).
```

Reply:
This suggestion should be incorporated in the latest force-push.